### ISSUE TITLE
aarch64: build broken 

### ISSUE THREAD

**Issue by chl337** [QuarkContainer/Quark#1244]
main does not build on aarch:
```
error[E0061]: this method takes 5 arguments but 4 arguments were supplied
    --> src/interrupt/aarch64/fault.rs:177:35
     |
177  |             let res = currTask.mm.InstallPageLocked(currTask, &vma, pageAddr, &range);
     |                                   ^^^^^^^^^^^^^^^^^---------------------------------- an ar
gument of type `bool` is missing
     |
note: method defined here
    --> src/qlib/kernel/memmgr/mm.rs:1049:12
```

The issues it seems trivial, but I am not sure how it was introduced.

HEAD: 1d3fbc23

**Comment by chl337** [comment]
Also, after building with #1245, I am noticing some regression on aarch64.

### Test
``` docker run --rm -it runtime=quark_d busybox sh``` + ``` ls ```
@shrik3 @CharlyYu can you confirm?

**Comment by shrik3** [comment]
On Fri May 17, 2024 at 12:37 PM CEST, chl337 wrote:
> Also, after building with #1245, I am noticing some regression on aarch64.
>
> ### Test
> ``` docker run --rm -it runtime=quark_d busybox sh``` + ``` ls ```
> @shrik3 @CharlyYu can you confirm?

as expected.

please follow up @1240 to apply the same optimization for aarch64 in #1245.
Adding the needs_write flag is not the whole story.

**Comment by chl337** [comment]
On Fri May 17, 2024 at 12:46 PM CEST, Tianhao Wang wrote:
> On Fri May 17, 2024 at 12:37 PM CEST, chl337 wrote:
> > Also, after building with #1245, I am noticing some regression on aarch64.
> >
> > ### Test
> > ``` docker run --rm -it runtime=quark_d busybox sh``` + ``` ls ```
> > @shrik3 @CharlyYu can you confirm?
>
> as expected.
>

> please follow up @1240 to apply the same optimization for aarch64 in #1245.
I am not sure the above is the right link:)

**Comment by shrik3** [comment]
> Also, after building with #1245, I am noticing some regression on aarch64.
> ### Test
> 
> ` docker run --rm -it runtime=quark_d busybox sh` + `ls` @shrik3 @CharlyYu can you confirm?

I may have a fix on that but what's your error here exactly?

**Comment by chl337** [comment]
> > Also, after building with #1245, I am noticing some regression on aarch64.
> > ### Test
> > ` docker run --rm -it runtime=quark_d busybox sh` + `ls` @shrik3 @CharlyYu can you confirm?
> 
> I may have a fix on that but what's your error here exactly?

` docker run --rm -it runtime=quark_d busybox sh` + `ls` :
❯ docker run --rm -it --runtime=quark_dlocal busybox sh
/ # ls
bin
c0e1e979-e79e-4781-b3a0-2a6d0ff6dc24
dev
etc
home
lib
lib64
proc
root
sys
tmp
usr
var
=> at this point, the kernel seems to have "crushed" for whatever reason, so it does not return to user land. IIRC we have faced some similar situation before, but only after extensively trying to reproduce it. Note, this does not happen for build in commands. I posted the log below for reference (no additional debug hints, sorry).

[quark.log](https://github.com/QuarkContainer/Quark/files/15355285/quark.log)
